Customizing/global/plugins/Services/Cron/CronHook/UserOrguImport/classes/Filesystem/FilesystemConfig.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace CaT\IliasUserOrguImport\Filesystem;
+
+/**
+ * Storage for filesystem related settings, i.e. deployment path.
+ */
+class FilesystemConfig
+{
+	const DEPLOYMENT_PATH = 'xuoi_data_deployment_path';
+
+	/**
+	 * @param	\ilSetting	$settings
+	 */
+	public function __construct($settings)
+	{
+		$this->settings = $settings;
+		$this->settings->read();
+	}
+
+	/**
+	 * Get currently configured deployment path of data source-files.
+	 *
+	 * @return	string|null
+	 */
+	public function deploymentPath()
+	{
+		$deployment_path = $this->settings->get(self::DEPLOYMENT_PATH);
+		if (trim((string)$deployment_path) !== '') {
+			return $deployment_path;
+		}
+		return null;
+	}
+
+	/**
+	 * Set deployment path of data source-files.
+	 *
+	 * @param	string	$path
+	 * @return	self
+	 */
+	public function withDeploymentPath($path)
+	{
+		assert('is_string($path)');
+		if (!is_readable($path) || !is_dir($path)) {
+			throw new \InvalidArgumentException($path.' may not be accessed');
+		}
+		$this->settings->set(self::DEPLOYMENT_PATH, $path);
```

Comment:
I don't like this. The name `with...` implies, that this is an immutable object, but in reality this mutates some internal value. Why's that?

Reply:
Ok,fixed.